pkg/authentication/authenticate.go

```diff
@@ -125,18 +136,35 @@ func (a *Authenticator[T]) Callback(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	id := uuid.NewString()
-	err = a.setSessionCookie(w, id)
-	if err != nil {
-		a.logger.Error("unable to save session cookie", "error", err, "id", id)
-		http.Error(w, "session could not be stored", http.StatusInternalServerError)
-		return
-	}
-	err = a.sessions.Set(id, ctx)
-	if err != nil {
-		a.logger.Error("unable to save session", "error", err, "id", id)
-		http.Error(w, "session could not be stored", http.StatusInternalServerError)
-		return
+	if a.useCookieSession {
```

Comment:
In go we try to avoid else blocks because it makes the code more verbose and early returns are easier to read (See https://go.dev/doc/effective_go#control-structures). So I would either refactor the if part to early return by adding

```go
http.Redirect(w, req, state.RequestedURI, http.StatusFound)
return
```

or create two functions (`statelessCallback` and `statefulCallback`)